Sasha congrats page

amak-code opened the pull request:
- Implemented **Congrats page** for **Intruder, CountWithMe and StoryFactoryLevel1 games**.
- Updated locize to have Spanish translated text for the Congratulations
- Updated StoryFactory logic, so after finishing SFL1 it goes to Congrats page and then back to Select page
- Updated DND.tsx to have a unique key for the DropTarget. Specifically added `letterIndex: number` to the `key={`${d.text}-${letterIndex}`}`

<img width="748" alt="Screen Shot 2024-11-12 at 10 37 32 AM" src="https://github.com/user-attachments/assets/6971d647-9152-4e5a-8248-e164d0a75324">

jonathan-chin commented:
@amak-code in the future, please try to group your PRs by feature.

the `DnD` edit here should be in a different PR so that the 2 groups of changes can be approved separately. the `DnD` edit is stale and was fixed in a different commit, so that change here needs to amended. if they were separate PRs, I could instead reject one and accept the other.

amak-code commented:
- Removed changes from DND.tsx
 - Updated StoryFactoryLevel2 and StoryFactoryCongrats so the SFL2 could show the Congrats page
 - StoryFactoryLevel2 still doesn't handle the number of attempts

jonathan-chin reviewed (approved):
overall, the file edits are good. however, there is a remaining bug that will need to be fixed in a future PR:

when clicking on "Keep Going" on SF Congrats, the user should be taken back to the existing SF game rather than back to pack select.
